Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/BoundsControl.cs

```diff
@@ -431,15 +427,32 @@ private void Awake()
             if (targetObject == null)
                 targetObject = gameObject;
 
+            scaleHandles = InstantiateScriptable(scaleHandles);
```

Comment:
I should probably not overwrite what we're serializing here, but apart from that I'm not sure if I'm instantiating this right. @Railboy , @Troy-Ferrell any hints?
From what I saw is that If I don't manually instantiate the instance of that scriptable will be shared across all the boundcontrols that link to it, which will result in all of them sharing the same gameobjects created for handles / boxdisplay etc. 
Am I supposed to instantiate them myself? I tried googling for this, but it seems people are using them purely as data storage?

Reply:
>From what I saw is that If I don't manually instantiate the instance of that scriptable will be shared across all the boundcontrols that link to it, which will result in all of them sharing the same gameobjects created for handles / boxdisplay etc.

This looks like it should work. It's an uncommon way to use ScriptableObjects. But I don't think it's improper.

> it seems people are using them purely as data storage?

Yes, they're typically just a raw data container and _maybe_ some utility functions. Assets like that don't need to be instantiated because there's no danger of cross-instance influence.

If we want to do things the 'standard' way then we would expect the BoundsControl instance to store any instance-specific objects and for the scriptable objects to manipulate those objects via static methods. Eg this:

```
// Game object used to display the box. Parented to the rig root
private GameObject boxDisplay;

internal void AddBoxDisplay(Transform parent, Vector3 currentBoundsExtents, FlattenModeType flattenAxis)
```

Becomes this:

```
internal static GameObject AddBoxDisplay(BoundsControlBoxDisplay display, Transform parent, Vector3 currentBoundsExtents, FlattenModeType flattenAxis)
```

Where the BoundsControl object is expected to keep a reference to boxDisplay and pass it to other methods, eg:

```       
internal static void SetHighlighted(GameObject boxDisplay)
```

@Troy-Ferrell thoughts on this approach?

Reply:
Regarding instantiation, yes you just create a copy at runtime.

Regarding the static functions and passing the instance target, I'd say either way. I don't see any clear advantage of one over the other. The static version is more verbose I guess

The bigger issue lies to me in the inspector
1) We need to make sure property manipulations are being done on the config file, not on this one gameobject instance...and more importantly that users understand this
2) When running play mode, we need to either disable the inspector or re-targeting the UI to the copied scriptable objects. Because changes made by the designer/user won't propagate.

Reply:
>I don't see any clear advantage of one over the other. The static version is more verbose I guess

The only real advantage is consistency. We'd be matching our config profile pattern, which people are already (sort of) comfortable with.

Reply:
I'd prefer not putting the local variables of each visual back into bounds control if anyhow avoidable. 

> We need to make sure property manipulations are being done on the config file, not on this one gameobject instance...and more importantly that users understand this

Agree on that. It might not be clear to people these are shared, especially if we're integrating them directly into the boundscontrol inspector gui
Apart from tooltips I can style that section different in the inspector so it's more visible it's not part of that object

> When running play mode, we need to either disable the inspector or re-targeting the UI to the copied scriptable objects. Because changes made by the designer/user won't propagate

I wonder if that already works as I'm already replacing the serialized field with the runtime instance in bounds control - not sure if the inspector would fetch the new value during runtime. I'll look into it!

Reply:
hmm the more i move stuff around the more i think I designed this wrong (ScriptableObject learning experience)- 
So I think it would be good to have both - a scriptable object as a shareable data container for all visuals AND a runtime created object that holds logic and instance relevant data members. 
So there's eg: BoundingBoxLinks which can be a simple c sharp class and BoundingBoxLinkConfiguration which is a Scriptable and holds all the immutable data. 
Bounding Box Setup for the user would be like this: 
![image](https://user-images.githubusercontent.com/36998103/67501639-8ccaf680-f67c-11e9-9dc7-a4e770f587ec.png)
So they can plug in and out their data sets for either the whole config or parts of the visual config. 
But those configs all stay separated, we just pass the data set to the logic classes but they won't modify anything.
It's similar to what Lars suggested with the static methods, just that we can still keep BoundsControl clean. 

Does that make sense?

Reply:
I've pushed a new version where the scriptable objects are just data containers and can be shared that way without the need of actually creating a new instance
further I moved the logic and runtime created objects into a separate class that is not part of the scriptable object and just reads the data from whatever is linked as scriptable configuration
However I didn't add a another configuration layer as scriptable to combine all the separate visual configs as sharing a whole configuration can also be achieved by using and inheriting from a prefab. In that case it's just making it more complicated for the user if I add another layer. Let me know what you think!